java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -453,24 +453,62 @@ public String uri() {
    *
    * @return the version of the dataset
    */
-  public long version() {
+  public Version getVersion() {
     try (LockManager.ReadLock readLock = lockManager.acquireReadLock()) {
       Preconditions.checkArgument(nativeDatasetHandle != 0, "Dataset is closed");
-      return nativeVersion();
+      return nativeGetVersion();
     }
   }
 
-  private native long nativeVersion();
+  private native Version nativeGetVersion();
+
+  /**
+   * Get the version history of the dataset.
+   *
+   * @return the version history of the dataset
+   */
+  public List<Version> listVersions() {
+    try (LockManager.ReadLock readLock = lockManager.acquireReadLock()) {
+      Preconditions.checkArgument(nativeDatasetHandle != 0, "Dataset is closed");
+      return nativeListVersions();
+    }
+  }
+
+  private native List<Version> nativeListVersions();
 
   /** @return the latest version of the dataset. */
-  public long latestVersion() {
```

Comment:
this will be backwards incompatible, can we avoid changing the name? (also the version to getVersion).

 I think one way we can do is:

``` 
public long version()
public Version versionDetail()
public long latestVersion()
public Version latestVersionDetail()
public List<Long> listVersions()
public List<Version> listVersionDetails()
```

what do you think?

Reply:
Yes.

This is the trade off between consistency api and compatible. I thought use more established convention getXXX in JAVA and kept these API one2one with python and rust would be elegent since we bring Version definition here.

The upper usage is mainly Spark(or somewhere I don't know? ), I could be resposible for that modification if we accept the incompatibility. I think Right now ​​the impact is controllable​. Just once for all.

If the incompatibility is more important. I also accept that

Reply:
I see. How about

```
//compatibility
public long version()
public long latestVersion()
//added
public Version getVersion()
public List<Version> listVersions()
```

The reduced two could be replaced by:

```
latestVersionDetail: checkoutLatest -> getVersion
listVersions(id): listVersions.map(Version::getId)
```